### PR TITLE
Add a section about the risks of priority and preemption

### PR DESCRIPTION
**Pull request by bsalamat**
Add a section about the risks of priority and preemption.

/sig scheduling

### CODE REVIEW

**Comment by bsalamat** [comment]
cc\ @davidopp

File: releases/release-1.11/release_notes_draft.md [hunk 1]
```diff
@@ -95,16 +95,32 @@ This tag is for changes that don't affect the user experience, such as behind-th
...
-[ResourceQuota](https://kubernetes.io/docs/concepts/policy/resource-quotas/)
-should be created to limit users from creating Pods at these priorities if not all users of your
-cluster are trusted. We do not advise disabling this feature since critical system Pods rely on
-the scheduler preemption to be scheduled when cluster is under resource pressure.
- 
+is enabled by default.\
+  Note that **it will be possible for users of the cluster to create pods that block some system daemons from running, and/or evict system daemons that are already running, by creating pods at the `system-cluster-critical` and `system-node-critical` priority classes, which are present in all clusters by default.** Please read the following information to understand the details. This is particularly important for those who have untrusted users in their Kubernetes clusters.
+
+  There are two kinds of critical system daemons in Kubernetes -- ones that run per-node as DaemonSets (e.g. fluentd, XXX list the rest of them here) and ones that run per-cluster (possibly more than one instance per cluster, but not one per node) (e.g. DNS, heapster, XXX list the rest of them here).
```
Comment (davidopp) [inline releases/release-1.11/release_notes_draft.md]: I guess this should say "There are two kinds of critical system pods" (not daemons)

Comment (davidopp) [inline releases/release-1.11/release_notes_draft.md]: Fix the first XXX to list the other node-level critical system pods and the second XXX to list the other cluster-level critical system pods.

File: releases/release-1.11/release_notes_draft.md [hunk 2]
```diff
@@ -95,16 +95,32 @@ This tag is for changes that don't affect the user experience, such as behind-th
...
+  
+  In Kubernetes 1.11, priority/preemption is enabled by default and
+  * per-node daemons continue to be scheduled directly by the DaemonSet controller, bypassing the default scheduler. As in Kubernetes versions before 1.11, the DaemonSet controller does not preempt pods, so we continue to rely on the ["rescheduler"](https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/) to guarantee that per-node daemons are able to schedule in a cluster that is full of regular user pods, by evicting regular user pods to make room for them. Per-node daemons are given a priority class of `system-node-critical`.
+  * cluster-level system pods continue to be scheduled by the default scheduler. The cluster-level daemons are given a priority class of `system-cluster-critical`. Because the default scheduler can preempt pods, the rescheduler in Kubernetes 1.11 is modified to *not* preempt pods to ensure the cluster-level system pods can schedule; instead we rely on the scheduler preemption mechanism to do this.
```
Comment (davidopp) [inline releases/release-1.11/release_notes_draft.md]: s/cluster-level daemons/cluster-level system pods/

File: releases/release-1.11/release_notes_draft.md [hunk 3]
```diff
@@ -95,16 +95,32 @@ This tag is for changes that don't affect the user experience, such as behind-th
...
+  * when cluster-level system pods are already running, a user can create a pod at `system-node-critical` priority and thereby preempt an already-running cluster-level system pod.
+  * a malicious user who fills the cluster with `system-node-critical` or `system-cluster-critical` pods can prevent cluster-level system pods from scheduling (note that they cannot prevent per-node daemons from scheduling, because the rescheduler ignores priority when deciding which pods it can evict, and it is still responsible for evicting pods to get per-node daemons to schedule in Kubernetes 1.11).
+
+  The only way to prevent this vulnerability is:
+  * Step 1: Configure the ResourceQuota admission controller (via a config file) to use the ["limitedResources"](https://kubernetes.io/docs/concepts/policy/resource-quotas/) feature to require quota for pods in PriorityClass `system-node-critical` and `system-cluster-critical`.
+  * Step 2: Enable the [`ResourceQuotaScopeSelectors`](https://kubernetes.io/docs/concepts/policy/resource-quotas/) feature gate (this is in alpha feature in Kubernetes 1.11)
```
Comment (davidopp) [inline releases/release-1.11/release_notes_draft.md]: s/is in/is an/

File: releases/release-1.11/release_notes_draft.md [hunk 4]
```diff
@@ -95,16 +95,32 @@ This tag is for changes that don't affect the user experience, such as behind-th
...
+  * Step 3: Create infinite ResourceQuota in the `kube-system` namespace at PriorityClass `system-node-critical` and `system-cluster-critical` using the [scopeSelector feature of ResourceQuota](https://kubernetes.io/docs/concepts/policy/resource-quotas/)
```
Comment (davidopp) [inline releases/release-1.11/release_notes_draft.md]: I guess this should say "infinite ResourceQuota for pods"

File: releases/release-1.11/release_notes_draft.md [hunk 5]
```diff
@@ -95,16 +95,32 @@ This tag is for changes that don't affect the user experience, such as behind-th
...
+
+  This will prevent anyone who does not have access to the `kube-system` namespace from creating pods with the `system-node-critical` or `system-cluster-critical` priority class, by only allowing pods with those priority classes to be created in the `kube-system` namespace.
```
Comment (davidopp) [inline releases/release-1.11/release_notes_draft.md]: the "by only allowing..." part could be a bit clearer: "by restricting pods with those priority classes to only be allowed in the `kube-system` namespace."

File: releases/release-1.11/release_notes_draft.md (hunk 1 above)
Comment (bsalamat) [reply releases/release-1.11/release_notes_draft.md]: Done.

Comment (bsalamat) [reply releases/release-1.11/release_notes_draft.md]: Done.

File: releases/release-1.11/release_notes_draft.md (hunk 2 above)
Comment (bsalamat) [reply releases/release-1.11/release_notes_draft.md]: done

File: releases/release-1.11/release_notes_draft.md (hunk 3 above)
Comment (bsalamat) [reply releases/release-1.11/release_notes_draft.md]: done

File: releases/release-1.11/release_notes_draft.md (hunk 4 above)
Comment (bsalamat) [reply releases/release-1.11/release_notes_draft.md]: done

File: releases/release-1.11/release_notes_draft.md (hunk 5 above)
Comment (bsalamat) [reply releases/release-1.11/release_notes_draft.md]: done

**Comment by bsalamat** [comment]
ping @nickchase @calebamiles for approval

**Comment by nickchase** [comment]
An abbreviated version of this has been added to the current doc, here: **https://docs.google.com/document/d/1MoHdmqSpWT4dJ3AcONwPwquNa2NIBa1dhpb0g8xyyoI/edit** with a link to this PR for the full story.  If someone's got a better idea, I'm all ears.

**Comment by davidopp** [comment]
The part you extracted seems fine, but please link to this PR rather than the one you are currently linking to.

**Comment by davidopp** [comment]
We'll need a new release note in 1.11.1 that explains the new admission controller that eliminates (for all practical purposes) the vulnerability.

**Comment by bsalamat** [comment]
@davidopp Sure. I will take care of that.

**Pull request closed without merging**